### PR TITLE
Keep dependencies dir on clean up

### LINKED ISSUE #218: Clone components only once
## Context

For each compilation of a cluster, commodore removes all components and clones them again. This is a waste of bandwidth and time.

Instead we could clone them into a separate directory. On compilation, commodore would only clone components not only present. For all the others, only a `git fetch` and `git checkout` will be required.

Making the components available within the inventory can then be done with symlinks.

### PR DESCRIPTION
**Pull request by corvus-ch**
Resolves #218

This one needs #250 to work properly.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update tests.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

### CODE REVIEW

File: commodore/helpers.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def clean_working_tree(config: Config):
         rmtree = shutil.rmtree
     click.secho("Cleaning working tree", bold=True)
     rmtree(config.inventory.inventory_dir, ignore_errors=True)
-    rmtree(config.inventory.dependencies_dir, ignore_errors=True)
```
Comment (srueg) [inline commodore/helpers.py:112]: This has the side-effect that Kapitan won't download HTTP(S) dependencies again if they already exist on disk. Which is ok-ish, I guess.

Should we maybe add a flag to the `cleanup` command to still be able to remove everything?

Comment (corvus-ch) [reply commodore/helpers.py:112]: No, in this case I just would also drop the `libs` dir. Downloading libs is not what takes time.

Comment (srueg) [reply commodore/helpers.py:112]: That's not what I meant. Kapitan dependencies are usually downloaded to `dependencies/component-name/manifests/name.yaml` or similar.

**Review by simu — APPROVED** [review]

**Review by srueg — APPROVED** [review]

**Comment by srueg** [comment]
Unfortunately this breaks if unstaged changes are present in dependencies:
```
cmdline: git checkout master
  stderr: 'error: Your local changes to the following files would be overwritten by checkout:
	class/resource-locker.yml
Please commit your changes or stash them before you switch branches.
Aborting'
```

**Comment by simu** [comment]
Hmm, I didn't cover that case in my `Component.checkout` tests. I assume this error happened when running a `catalog compile` without the `--local` flag.

**Comment by srueg** [comment]
Yes exactly. I think it's actually a nice thing because it prevents local changes from being lost. but maybe the error message could be a little nicer and we could introduce a `--force/-f` flag for example to force a checkout?

**Comment by simu** [comment]
Yes, I agree, we need to properly handle this case and produce a user-friendly error. And a force flag to throw away changes probably makes sense too

**Pull request merged**